### PR TITLE
Refactor backend code to use Typescript

### PR DESCRIPTION
**Pull request by Mapleia**
## Changes
Installed Typescript and refactored code to include typing. Also installed eslint.

## Requirements
Requires: 
- `typescript` dev package
- `cors` dev types
- `express` dev types
- `jest` dev types
- `node` dev types
- `typescript` dev eslint dependency

## PR Dependencies
Depends on: #9 

## Subtasks / Progress
- [x] Change Dockerfile so that typescript build runs.

## Testing
- [ ] Manual 
- [x] Automated

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,6 +4,9 @@
   "description": "Services for Bipolar Bridges Project",
   "main": "main.js",
   "scripts": {
+    "tsc:api": "cd services/http/api && yarn build",
+    "tsc:mock": "cd services/http/api && yarn build:mock",
```
Comment (rlnsy) [inline package.json]: can we change the prefix to 'build' instead of 'tsc'?

Comment (rlnsy) [reply package.json]: Also, should clarify if mock refers to mock api or mock auth server

File: package.json [hunk 2]
```diff
@@ -4,6 +4,9 @@
...
+    "lint": "eslint . --ext .ts",
```
Comment (rlnsy) [inline package.json]: Can we also add this command to the api directory?

File: services/http/api/package.json [hunk 3]
```diff
@@ -1,10 +1,12 @@
...
   "name": "bb-api",
   "description": "Internal API for Bipolar Bridges Project",
-  "main": "main.js",
+  "main": "/dist/main.js",
   "scripts": {
-    "test": "jest tests",
-    "start": "node main.js",
+    "build": "rm -rf ./dist && tsc",
+    "build:mock": "cd tests/mocks/auth_server && yarn build",
+    "test": "jest dist/tests",
```
Comment (rlnsy) [inline services/http/api/package.json]: we should look into using tsjest instead (see companion app repo for how to do this, specifically the testing workflow PR)

**Review by rlnsy — CHANGES_REQUESTED** [review]
Looks good so far. I haven't looked into the actual code too much yet, since it's hard to see the diffs. in github. See my comments for now and I'll do another pass soon.

File: package.json (hunk 1 above)
Comment (Mapleia) [reply package.json]: Sounds good. Will do!

File: package.json (hunk 2 above)
Comment (Mapleia) [reply package.json]: Can do! Need to move the eslint file around, but should be not too hard to do.

File: services/http/api/package.json (hunk 3 above)
Comment (Mapleia) [reply services/http/api/package.json]: Just added and configured ts-jest!

File: services/http/api/.eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    "parser": "@typescript-eslint/parser",
+    "parserOptions": {
+        "ecmaVersion": 12,
+        "sourceType": "module"
+    },
+    "plugins": [
+        "@typescript-eslint"
+    ],
+    "rules": {
+        "no-explicit-any": 0,
```
Comment (rlnsy) [inline services/http/api/.eslintrc.js]: I think I'd prefer to use no-implicit-any instead

**Review by rlnsy — DISMISSED** [review]
Just the one comment about lint rules - otherwise looks great!

File: services/http/api/.eslintrc.js (hunk 4 above)
Comment (Mapleia) [reply services/http/api/.eslintrc.js]: I think I fixed up the issues with explicit any anyways, so I think it's safe to remove.

**Review by wynnset — DISMISSED** [review]

**Pull request merged**